util/request_source.go

```diff
@@ -0,0 +1,67 @@
+package util
+
+import (
+	"context"
+)
+
+// RequestSourceTypeKeyType is a dummy type to avoid naming collision in context.
+type RequestSourceTypeKeyType struct{}
+
+// RequestSourceTypeKey is used as the key of request source type in context.
+var RequestSourceTypeKey = RequestSourceTypeKeyType{}
+
+const (
+	// InternalTxnOthers is the type of requests that consume low resources.
+	// This reduces the size of metrics.
+	InternalTxnOthers = "others"
+	// InternalTxnGC is the type of GC txn.
+	InternalTxnGC = "gc"
+	// InternalTxnMeta is the type of the miscellaneous meta usage.
+	InternalTxnMeta = InternalTxnOthers
```

Comment:
Some sources are client's inner types and applications use client can give their specified types. Also the types are aliased to reduce the metrics.